### PR TITLE
Nc news 3

### PR DESCRIPTION
**Pull request by joeglDev**
added GET /api/topics functionality for Ticket 3 + invalid path error handling.

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    return request(app).get(Endpoints.ALL_TOPICS_END).expect(200);
+  });
+  test("endpoint responds with an array of length 3", () => {
+    return request(app)
+      .get(Endpoints.ALL_TOPICS_END)
+      .then(({ body: { topics } }) => {
+        expect(topics.length).toBe(3);
+      });
+  });
+  test("invalid path returns a http status code of 404", () => {
```
Comment (k-huggs) [inline __tests__/endpoints.test.js:24]: this has potential to be conflated with a 400 error with an endpoint with an invalid end i.e.
/api/topics/invalid - 400

**Comment by k-huggs** [comment]
Code overall is fine, couple comments for consideration. feel free to merge

File: __tests__/endpoints.test.js (hunk 1 above)
Comment (joeglDev) [reply __tests__/endpoints.test.js:24]: next request will change to http status: 400

File: app.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+//const port = 9090;
+
+const app = express();
+module.exports = app;
+
+//Endpoints
+app.get(Endpoints.ALL_TOPICS_END, getAllTopics);
+
+//Error handling
+app.get('*', handleInvalidPaths);
```
[the remark this answers is not in the document]
Comment (joeglDev) [reply app.js:15]: app.get -> app.use

**Pull request closed without merging**